src/network-area-diagram-viewer.ts

```diff
@@ -0,0 +1,148 @@
+        return this.height;
+    }
+
+    public getContainer(): HTMLElement {
+        return this.container;
+    }
+
+    public getSvgContent(): string {
+        return this.svgContent;
+    }
+
+    public init(
+        minWidth: number,
+        minHeight: number,
+        maxWidth: number,
+        maxHeight: number
+    ): void {
+        if (
+            this.container === null ||
+            this.container === undefined ||
+            this.isSvgContentValid()
+        ) {
+            return;
+        }
+
+        const dimensions: DIMENSIONS = this.getDimensionsFromSvg();
+
+        if (dimensions === null) {
+            return;
+        }
+
+        // clear the previous svg in div element before replacing
+        this.container.innerHTML = '';
+
+        this.setWidth(
+            dimensions.width < minWidth
+                ? minWidth
+                : Math.min(dimensions.width, maxWidth)
+        );
+        this.setHeight(
+            dimensions.height < minHeight
+                ? minHeight
+                : Math.min(dimensions.height, maxHeight)
+        );
+        const draw = SVG()
+            .addTo(this.container)
+            .size(this.width, this.height)
+            .viewbox(
+                dimensions.viewbox.x,
+                dimensions.viewbox.y,
+                dimensions.viewbox.width,
+                dimensions.viewbox.height
+            )
+            .panZoom({
+                panning: true,
+                zoomMin: 0.5,
+                zoomMax: 200,
+                zoomFactor: 0.3,
+                margins: { top: 0, left: 0, right: 0, bottom: 0 },
+            });
+
+        const drawnSvg: HTMLElement = <HTMLElement>(
+            draw.svg(this.svgContent).node.firstElementChild
+        );
+        drawnSvg.style.overflow = 'visible';
+        // PowSyBl NAD introduced server side calculated SVG viewbox. This viewBox attribute can be removed as it is copied in the panzoom svg tag.
+        (<HTMLElement>draw.node.firstChild).removeAttribute('viewBox');
+        (<HTMLElement>draw.node.firstChild).removeAttribute('width');
+        (<HTMLElement>draw.node.firstChild).removeAttribute('height');
+    }
+
+    public getDimensionsFromSvg(): DIMENSIONS {
+        // Dimensions are set in the main svg tag attributes. We want to parse those data without loading the whole svg in the DOM.
+        const result = this.svgContent.match('(<svg)(.*)(>)');
+        if (result === null || result.length === 0) {
+            return null;
+        }
+        const emptiedSvgContent = result[0];
+        const svg: SVGSVGElement = new DOMParser()
+            .parseFromString(emptiedSvgContent, 'image/svg+xml')
+            .getElementsByTagName('svg')[0];
+        const width: number = +svg.getAttribute('width');
+        const height: number = +svg.getAttribute('height');
+        const viewbox: VIEWBOX = svg.viewBox.baseVal;
+        return { width: width, height: height, viewbox: viewbox };
```

Comment:
Add a test ?

Reply:
Svg manipulation isn't well supported with Jest. The test doesn't pass.